src/test/java/com/github/lwhite1/tablesaw/TableTest.java

```diff
@@ -14,35 +22,61 @@
  */
 public class TableTest {
 
-  private Table table;
-  private FloatColumn column = new FloatColumn("f1");
-
-  @Before
-  public void setUp() throws Exception {
-    table = new Table("t");
-    table.addColumn(column);
-  }
-
-  @Test
-  public void testColumn() throws Exception {
-    Column column1 = table.column(0);
-    assertNotNull(column1);
-  }
-
-  @Test
-  public void testColumnCount() throws Exception {
-    assertEquals(0, new Table("t").columnCount());
-    assertEquals(1, table.columnCount());
-  }
-
-  @Test
-  public void testRowCount() throws Exception {
-    assertEquals(0, table.rowCount());
-    FloatColumn floatColumn = column;
-    floatColumn.add(2f);
-    assertEquals(1, table.rowCount());
-
-    floatColumn.add(2.2342f);
-    assertEquals(2, table.rowCount());
-  }
+    private Table table;
+    private FloatColumn column = new FloatColumn("f1");
+
+    @Before
+    public void setUp() throws Exception {
+        table = new Table("t");
+        table.addColumn(column);
+    }
+
+    @Test
+    public void testColumn() throws Exception {
+        Column column1 = table.column(0);
+        assertNotNull(column1);
+    }
+
+    @Test
+    public void testColumnCount() throws Exception {
+        assertEquals(0, new Table("t").columnCount());
+        assertEquals(1, table.columnCount());
+    }
+
+    @Test
+    public void testRowCount() throws Exception {
+        assertEquals(0, table.rowCount());
+        FloatColumn floatColumn = column;
+        floatColumn.add(2f);
+        assertEquals(1, table.rowCount());
+
+        floatColumn.add(2.2342f);
+        assertEquals(2, table.rowCount());
+    }
+
+    @Test
+    public void testGetRow() throws Exception {
+
+        Table table = TestData.SIMPLE_UNSORTED_DATA.getTable();
+        Path source = TestData.SIMPLE_UNSORTED_DATA.getSource();
+        List<String[]> rawData;
+        try (CSVReader csvReader = new CSVReader(Files.newBufferedReader(source, Charset.defaultCharset()))) {
+            rawData = csvReader.readAll();
+        }
+        // remove the first row from the raw data since tables don't keep column header row as part of their data
+        rawData.remove(0);
+        int maxRows = table.rowCount();
+        for (int rowIndex = 0; rowIndex < maxRows; rowIndex++) {
+            String row = table.getRow(rowIndex);
+            String expectedRow = getCSVfromListOfStrings(rawData.get(rowIndex));
+            assertEquals("Row " + rowIndex + " matches", expectedRow,row);
+        }
```

Comment:
sorry. that should have read:

you can't expect it to give you back the date as a String in a specific
format unless you somehow specify the format. Tablesaw uses ISO format to
output Strings from dates by default, but if you need a specific format you
can get a Java 8 localDate from the column and use standard date formatting.

But please see my comment on why you don't want to write this method
without using OpenCSV (and don't really want it in the Table class.)

On Sun, Jun 19, 2016 at 9:46 AM, Larry White <ljw1001@gmail.com> wrote:

> you can't expect it to give you back the date as a String in a specific
> format unless you somehow specify the format. Tablesaw uses ISO format to
> output Strings from dates by default, but if you need a specific format you
> can get a Java 8 localDate from the column and use standard date formatting.
>
> to get a java 8 LocalDate object, then string format that any way you
> want.  But please see my comment on why you don't want to write this method
> without using OpenCSV (and don't really want it in the Table class.)
>
>
>
> On Sun, Jun 19, 2016 at 9:40 AM, Spencer Marks <notifications@github.com>
> wrote:
>
>> In src/test/java/com/github/lwhite1/tablesaw/TableTest.java
>> <https://github.com/lwhite1/tablesaw/pull/19#discussion_r67617181>:
>>
>> > +    public void testGetRow() throws Exception {
>> > +
>> > +        Table table = TestData.SIMPLE_UNSORTED_DATA.getTable();
>> > +        Path source = TestData.SIMPLE_UNSORTED_DATA.getSource();
>> > +        List<String[]> rawData;
>> > +        try (CSVReader csvReader = new CSVReader(Files.newBufferedReader(source, Charset.defaultCharset()))) {
>> > +            rawData = csvReader.readAll();
>> > +        }
>> > +        // remove the first row from the raw data since tables don't keep column header row as part of their data
>> > +        rawData.remove(0);
>> > +        int maxRows = table.rowCount();
>> > +        for (int rowIndex = 0; rowIndex < maxRows; rowIndex++) {
>> > +            String row = table.getRow(rowIndex);
>> > +            String expectedRow = getCSVfromListOfStrings(rawData.get(rowIndex));
>> > +            assertEquals("Row " + rowIndex + " matches", expectedRow,row);
>> > +        }
>>
>> so there is a bug in this data. TableSaw converts date format
>>
>> org.junit.ComparisonFailure: Row 0 matches
>> Expected :Spencer,52,Harvard,9/4/63
>> Actual :Spencer,52,Harvard,2063-09-04
>>
>> which is interesting.
>>
>> —
>> You are receiving this because you commented.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/lwhite1/tablesaw/pull/19/files/cd380a70fe6a53499b7d3ac576d609ae1dc77932#r67617181>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/ADRXgjnyQxYH42AfaLLCI-cs4KtSk5ylks5qNUa_gaJpZM4I5ChM>
>> .
>>
>
>